### PR TITLE
Telegram long polling

### LINKED ISSUE #561: telegram polling protocol environment
As I noticed before about private server where running Grafana OnCall need use ChatOps/Alert notifications through webhook with self-signed certificate (https://github.com/grafana/oncall/issues/530), but what about polling protocol? 

I think if we get telegram connections through polling protocol it will be more useful _than we trying connect by webhook from private servers_. In results there's no need to open external port for receiving messages from telegram bot. :)

### PR DESCRIPTION
**Pull request by alexintech**
# What this PR does

Runs Telegram long polling to get updates. 
It's enabled by setting `FEATURE_TELEGRAM_LONG_POLLING_ENABLED=True`. That will disable webhook and run separate deployment for telegram long polling.

Telegram long polling is not very HA mode, but it does not need to expose webhook url to internet and simplifies telegram integration.

## Which issue(s) this PR fixes

closes #561 

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: dev/.env.dev.example [hunk 1]
```diff
@@ -28,7 +28,7 @@ SOCIAL_AUTH_REDIRECT_IS_HTTPS=False
 GRAFANA_INCIDENT_STATIC_API_KEY=
 GRAFANA_API_URL=http://localhost:3000
 
-CELERY_WORKER_QUEUE="default,critical,long,slack,telegram,webhook,retry,celery"
+CELERY_WORKER_QUEUE=default,critical,long,slack,telegram,webhook,retry,celery
```
Comment (alexintech) [inline dev/.env.dev.example:31]: removed quotes, as I found that starting `./celery_with_exporter.sh` locally will create queues `"default`, `celery"` with quotes.

File: engine/settings/base.py [hunk 2]
```diff
@@ -65,6 +65,7 @@
 FEATURE_PROMETHEUS_EXPORTER_ENABLED = getenv_boolean("FEATURE_PROMETHEUS_EXPORTER_ENABLED", default=False)
 GRAFANA_CLOUD_ONCALL_HEARTBEAT_ENABLED = getenv_boolean("GRAFANA_CLOUD_ONCALL_HEARTBEAT_ENABLED", default=True)
 GRAFANA_CLOUD_NOTIFICATIONS_ENABLED = getenv_boolean("GRAFANA_CLOUD_NOTIFICATIONS_ENABLED", default=True)
+TELEGRAM_LONG_POLLING_ENABLED = getenv_boolean("TELEGRAM_LONG_POLLING_ENABLED", default=False)
```
Comment (alexintech) [inline engine/settings/base.py:68]: maybe it should be named `FEATURE_TELEGRAM_LONG_POLLING_ENABLED`?

File: Makefile [hunk 3]
```diff
@@ -18,6 +18,7 @@ REDIS_PROFILE = redis
 RABBITMQ_PROFILE = rabbitmq
 PROMETHEUS_PROFILE = prometheus
 GRAFANA_PROFILE = grafana
+TELEGRAM_POLLING_PROFILE = telegram_polling
```
Comment (joeyorlando) [inline Makefile:21]: 👋 do you mind adding this new compose profile [here](https://github.com/grafana/oncall/blob/dev/dev/README.md#compose_profiles:~:text=The%20possible%20profiles%20values%20are%3A)?

File: docker-compose-developer.yml [hunk 4]
```diff
@@ -82,6 +82,22 @@ services:
     profiles:
       - engine
 
+  oncall_telegram_polling:
```
Comment (joeyorlando) [inline docker-compose-developer.yml:85]: is this new container missing the following?
```yaml
user: *oncall-user
```

Comment (alexintech) [reply docker-compose-developer.yml:85]: done

**Comment by alexintech** [comment]
👀

**Comment by joeyorlando** [comment]
@alexintech thanks for the contribution (again 😄).

I'm not very familiar w/ Telegram. I'll ask around and see if someone else more familiar can review this.

**Comment by Sammyant** [comment]
Hi everyone. what about this MR? 
we so need this cool feature to avoid publishing our grafana on call

**Comment by gproxyz** [comment]
Hi everyone. 
We need this PR. 
pls

**Comment by EsDmitrii** [comment]
Hi!
Awesome feature! When do you plan to release this?
We need that :)

File: engine/settings/base.py [hunk 5]
```diff
@@ -67,6 +67,7 @@
 FEATURE_GRAFANA_ALERTING_V2_ENABLED = getenv_boolean("FEATURE_GRAFANA_ALERTING_V2_ENABLED", default=False)
 GRAFANA_CLOUD_ONCALL_HEARTBEAT_ENABLED = getenv_boolean("GRAFANA_CLOUD_ONCALL_HEARTBEAT_ENABLED", default=True)
 GRAFANA_CLOUD_NOTIFICATIONS_ENABLED = getenv_boolean("GRAFANA_CLOUD_NOTIFICATIONS_ENABLED", default=True)
+TELEGRAM_LONG_POLLING_ENABLED = getenv_boolean("TELEGRAM_LONG_POLLING_ENABLED", default=False)
```
Comment (matiasb) [inline engine/settings/base.py:70]: I think you missed to rename the setting here? (it should be `FEATURE_TELEGRAM_LONG_POLLING_ENABLED` instead, right? and the env variable name as well)

File: helm/oncall/templates/_env.tpl [hunk 6]
```diff
@@ -95,9 +95,16 @@
 {{- end }}
 
 {{- define "snippet.oncall.telegram.env" -}}
+{{- if .Values.telegramPolling.enabled -}}
+{{- $_ := set .Values.oncall.telegram "enabled" true -}}
```
Comment (matiasb) [inline helm/oncall/templates/_env.tpl:99]: Does this mean that setting the telegram polling flag will "automatically" turn on the telegram enabled value?

**Review by matiasb — COMMENTED** [review]
Looks good to me :+1: , minor comments added.

Maybe @iskhakov can take a look at the helm updates? (they make sense to me, but to get another couple of eyes)

File: helm/oncall/templates/_env.tpl (hunk 6 above)
Comment (alexintech) [reply helm/oncall/templates/_env.tpl:99]: Yes, exactly.

File: engine/settings/base.py (hunk 5 above)
Comment (alexintech) [reply engine/settings/base.py]: You're right. Fixed name, and moved it up next to `FEATURE_TELEGRAM_INTEGRATION_ENABLED` setting

File: helm/oncall/templates/telegram-polling/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+          command: ['sh', '-c', 'python manage.py start_telegram_polling']
+          env:
+            {{- include "snippet.oncall.env" . | nindent 12 }}
+            {{- include "snippet.oncall.telegram.env" . | nindent 12 }}
+            {{- if eq .Values.database.type "mysql" }}
+            {{- include "snippet.mysql.env" . | nindent 12 }}
+            {{- end }}
+            {{- if eq .Values.database.type "postgresql" }}
+            {{- include "snippet.postgresql.env" . | nindent 12 }}
+            {{- end }}
```
Comment (joeyorlando) [inline helm/oncall/templates/telegram-polling/deployment.yaml:47]: similar to below comment, [new snippet](https://github.com/grafana/oncall/pull/2751/files#diff-e3f09fcc93f2a991e715f7c7908075e5ffc64d4ed1ac806c389f279b43bfb596R534-R544): 
```suggestion
            {{- include "snippet.db.env" . | nindent 12 }}
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -274,7 +280,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - UI drawer updates for webhooks2 ([#2419](https://github.com/grafana/oncall/pull/2419))
-- Removed url from sms notification, changed format ([#2317](https://github.com/grafana/oncall/pull/2317))
+- Removed url from sms notification, changed format ([2317](https://github.com/grafana/oncall/pull/2317))
```
Comment (joeyorlando) [inline CHANGELOG.md:283]: are these two changes intentional?

Comment (alexintech) [reply CHANGELOG.md]: I think, they accidentally appeared after merge

File: helm/oncall/tests/telegram_polling_deployment_test.yaml [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+      - equal:
+          path: spec.template.spec.containers[0].name
+          value: telegram-polling
+      - equal:
+          path: spec.template.spec.containers[0].command
+          value: ['sh', '-c', 'python manage.py start_telegram_polling']
+      - notExists:
+          path: spec.template.spec.containers[0].resources
+      - matchSnapshot:
+          path: spec.template.spec.containers[0].env
```
[the remark this answers is not in the document]
Comment (alexintech) [reply helm/oncall/tests/telegram_polling_deployment_test.yaml]: changed to use `matchSnapshot` for containers. I don't want to use snapshot for `spec` as in that case snapshot will be too big. I think very big snapshots are difficult to read and to understand, if you need for some reason.

**Review by joeyorlando — APPROVED** [review]

**Pull request merged**